src/com/amazon/ion/impl/IonReaderBinaryRawX.java

```diff
@@ -876,36 +876,15 @@ protected final long readULong(int len) throws IOException
     {
         long    retvalue = 0;
         int b;
-        switch (len) {
-        default:
-            throw new IonException("value too large for Java long");
-        case 8:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 7:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 6:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 5:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 4:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 3:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 2:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 1:
-            if ((b = read()) < 0) throwUnexpectedEOFException();
-            retvalue = (retvalue << 8) | b;
-        case 0:
-            // do nothing, it's just a 0 length is a 0 value
-        }
+        if(len == 0)
+            {
+                //do nothing its just 0[NULl]
+            }
+        else
+            {
+                if ((b = read()) < 0) throwUnexpectedEOFException();
```

Comment:
The current implementation relies on [`switch` statement fall-through ](https://en.wikipedia.org/wiki/Switch_statement#Fallthrough) to read and process the correct number of bytes. This new code is not functionally equivalent, which is why [the CI build is failing](https://travis-ci.org/github/amzn/ion-java/builds/719974334).